client/src/todoAppComponents/todos/TodoDetails.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { TodoDetailsContainer } from "../StyledItems";
+import { selectItem } from "../../store/actions/todos";
+import * as api from "../../http/api";
+
+const TodoDetails = () => {
+
+    const dispatch = useDispatch();
+    const id = useSelector(state => state.todos.selectedItem);
+    const [ todoInfo, setTodoInfo ] = useState(null);
+
+    const onClose = () => {
+        dispatch(selectItem(null));
+    };
+
+    useEffect(() => {
+        const fetchData = async() => {
+            return await api.fetchTodoDetailsById(id);
+        };
+        if (id) {
+            fetchData().then(({ data }) => setTodoInfo(data)) ;
+        }
+        return () => {
+            setTodoInfo(null);
+        }
```

Comment:
check if the component is still mounted (in case a user left the page before the request was completed)
- add ref and set it to true/ false and check in useEffect the value